WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift

```diff
@@ -109,36 +79,26 @@ final class TotalsViewModel: ObservableObject, TotalsViewModelProtocol {
     func stopShowingTotalsView() {
         posModel.cancelCardReaderPreparation()
     }
+
+    func shouldShowTotalsFields(for paymentState: PointOfSalePaymentState) -> Bool {
```

Comment:
Yes – but it's also important to have the `onChange` handler which observes this (as we do.) 

If it didn't work in the `onChange`,  we would continue to see the totals fields in the `Processing payment` and `Payment successful` steps, which are also handled by this (over-narrowly named!) view.

From my research, I'm not totally clear on how `onChange` detects a possible change in the value of this computed var, so you're right to bring it up. The var relies on the view model, which (in later PRs) doesn't have any published properties.

I think it's working because when the `paymentState` changes the view is invalidated due to the changes in `posModel.paymentState` (a published property which it also observes.) At that point the `onChange` is evaluated against the old value, which calls the computed var, and recognises it as a change, so shows or hides the totals fields.

This might be a side-effect, but it's tricky to prove. As we move towards `Observation`, this is the sort of thing we'll need to keep a close eye on, because the more granular view updates could result in these state changes breaking. Better sooner than later!